File: LeetCodeCPP/1207.unique-number-of-occurrences.cpp
/*
 * @lc app=leetcode id=1207 lang=cpp
 *
 * [1207] Unique Number of Occurrences
 */

#include <vector>
#include <unordered_map>
#include <unordered_set>

using namespace std;

// @lc code=start
class Solution
{
public:
    bool uniqueOccurrences(vector<int> &arr)
    {
        unordered_map<int, int> freqMap;
        for (auto num : arr)
        {
            freqMap[num]++;
        }

        unordered_set<int> freqSet;
        for (auto [key, value] : freqMap)
        {
            freqSet.insert(value);
        }

        return freqMap.size() == freqSet.size();
    }
};
// @lc code=end

File: LeetCodePython/1207.unique-number-of-occurrences.py
#
# @lc app=leetcode id=1207 lang=python3
#
# [1207] Unique Number of Occurrences
#

from typing import List
from collections import Counter


# @lc code=start
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        freq_dict = Counter(arr)

        freq_set = set()

        for v in freq_dict.values():
            freq_set.add(v)

        return len(freq_set) == len(freq_dict)


# @lc code=end
